### ISSUE TITLE
Question about c4 pretraining

### ISSUE THREAD

**Issue by ontocord** [google-research/text-to-text-transfer-transformer#550]
Am I understanding the pre-training setup properly in that C4 is used only to train the decoder, in a causal language model? While the other supervised tasks are used to train the encoder and decoder. So the encoder is never trained on C4?

```

# ==================================== C4 ======================================
# Final pretraining task used in Raffel et al., 2019.
TaskRegistry.add(
    "c4_v220_span_corruption",
    TfdsTask,
    tfds_name="c4/en:2.2.0",
    text_preprocessor=functools.partial(
        preprocessors.rekey, key_map={"inputs": None, "targets": "text"}),
    token_preprocessor=preprocessors.span_corruption,
    output_features=DEFAULT_OUTPUT_FEATURES,
    metric_fns=[])
```

This was confusing as the paper seem to imply that you do an encoder decoder training, using a denoising dataset, as shown in Fig 2.

**Comment by craffel** [comment]
No, the pre-training includes both the encoder and decoder. Masked input sequences are fed into the encoder, and the decoder fills in the masked tokens by attending to the encoder output.

**Comment by ontocord** [comment]
Got it. Thanks @craffel
